src/front/wgsl/mod.rs

```diff
@@ -2305,10 +2233,14 @@ impl Parser {
                 self.pop_scope(lexer);
                 expr
             }
-            token @ (Token::Word("true" | "false") | Token::Number { .. }, _) => {
-                let _ = lexer.next();
-                let const_handle =
-                    self.parse_const_expression_impl(token, lexer, None, ctx.types, ctx.constants)?;
+            (Token::Word("true" | "false") | Token::Number(..), _) => {
+                let const_handle = self.parse_const_expression_impl(
+                    lexer.next(),
+                    lexer,
+                    None,
+                    ctx.types,
+                    ctx.constants,
+                )?;
```

Comment:
I think this is just `self.parse_const_expression(lexer, None, ctx.types, ctx.constants)`.